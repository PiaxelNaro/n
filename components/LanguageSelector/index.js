// @flow
import React from 'react';
import invert from 'lodash/invert';

import { LANG_MAP } from '../../modules/i18n';
import TagList from '../TagList';

const ISO_LANG_MAP: Object = invert(LANG_MAP);

const onLanguageChange = (lang: string) => {
    const location = global.location;
    const pathname = location.pathname;
    const search = location.search;
    const hash = location.hash;
    const next = encodeURIComponent(`${pathname}${search}${hash}`);
    const shorthandLang = ISO_LANG_MAP[lang];
    const url = `/i18n/activate/${shorthandLang}/?next=${next}`;

    location.assign(url);
};

const LanguageSelector = ({
    current,
    languages,
}: {
    current: string,
    languages: Array<{
        code: string,
        label: string,
    }>,
}) =>
    <TagList
        items={languages.map(({ code, label }) => ({
            id: code,
            name: label,
        }))}
        current={current}
        onChange={onLanguageChange}
    />;

export default LanguageSelector;
