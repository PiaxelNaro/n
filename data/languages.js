export default [
    {
        code: 'en-US',
        label: 'English',
    },
    {
        code: 'ja-JP',
        label: '日本語',
    },
    {
        code: 'ko-KR',
        label: '한국어',
    },
    {
        code: 'zh-CN',
        label: '中文(简体)',
    },
    {
        code: 'ru-RU',
        label: 'Русский',
    },
    {
        code: 'fr-FR',
        label: 'Français',
    },
    {
        code: 'de-DE',
        label: 'Deutsch',
    },
];
