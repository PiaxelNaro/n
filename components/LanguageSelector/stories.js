// @flow
import React from 'react';
import { storiesOf } from '@kadira/storybook';
import LanguageSelector from '.';
import languages from '../../data/languages';

storiesOf('LanguageSelector', module).add('render language selector', () =>
    <LanguageSelector languages={languages} current={'en-US'} />
);
