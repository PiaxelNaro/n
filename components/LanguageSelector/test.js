// @flow
import React from 'react';
import { shallow } from 'enzyme';
import LanguageSelector from '.';

import languages from '../../data/languages';

describe('<LanguageSelector>', () => {
    it('renders languages', () => {
        global.location.assign = jest.fn();
        const wrapper = shallow(
            <LanguageSelector languages={languages} current="en-US" />
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.find('TagList').prop('onChange')('ja-JP');
        expect(global.location.assign).toHaveBeenCalled();
    });
});
