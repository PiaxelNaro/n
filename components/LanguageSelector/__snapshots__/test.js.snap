// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelector> renders languages 1`] = `
<TagList
  current="en-US"
  items={
    Array [
      Object {
        "id": "en-US",
        "name": "English",
      },
      Object {
        "id": "ja-JP",
        "name": "日本語",
      },
      Object {
        "id": "ko-KR",
        "name": "한국어",
      },
      Object {
        "id": "zh-CN",
        "name": "中文(简体)",
      },
      Object {
        "id": "ru-RU",
        "name": "Русский",
      },
      Object {
        "id": "fr-FR",
        "name": "Français",
      },
      Object {
        "id": "de-DE",
        "name": "Deutsch",
      },
    ]
  }
  onChange={[Function]}
/>
`;
